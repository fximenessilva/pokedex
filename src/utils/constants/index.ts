const BASE_URL = 'https://pokeapi.co/api/v2';

const API_ENDPOINTS = {
  pokemons_list: BASE_URL + '/pokemon?limit=151',
  pokemon_detail: (name: string) => `${BASE_URL}/pokemon/${name}`,
} as const;

const IMAGES_URL = (name: string | undefined) => {
  if (!name) return;
  return `https://img.pokemondb.net/sprites/black-white/anim/normal/${name}.gif`;
};

const LOGO_URL =
  'https://upload.wikimedia.org/wikipedia/commons/9/98/International_Pok%C3%A9mon_logo.svg';

const POKEBALL_URL =
  'https://cdn.pnghd.pics/data/804/pokeball-transparent-png-30.png';

const BREAKPOINTS = {
  xsmall: '360px',
  small: '576px',
  medium: '768px',
  large: '992px',
  xlarge: '1200px',
} as const;

const NAMESPACES = {
  theme: 'pokedex-theme-dark',
  favorites: 'favorites_pokemons',
} as const;

const COLORS = {
  yellow: '#FFD700',
  light_gray: '#F5F5F5',
  lighter_gray: '#E0E0E0',
  gray: '#808080',
  medium_gray: '#CCCCCC',
  dark_gray: '#888888',
  darker_gray: '#666666',
  darkest_gray: '#333333',
  black: '#000000',
  green: '#4CAF50',
  blue: '#003A70',
} as const;

const ROUTES = {
  homepage: '/',
  pokemons_list: '/pokemon',
  pokemons_favorites: '/favorite_pokemons',
  pokemon_detail: '/pokemon/:pokemonId',
} as const;

export {
  API_ENDPOINTS,
  BASE_URL,
  IMAGES_URL,
  LOGO_URL,
  BREAKPOINTS,
  NAMESPACES,
  COLORS,
  ROUTES,
  POKEBALL_URL,
};
