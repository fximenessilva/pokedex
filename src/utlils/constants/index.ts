const BASE_URL = 'https://pokeapi.co/api/v2';

const API_ENDPOINTS = {
  pokemons_list: BASE_URL + '/pokemon?limit=151',
  pokemon_detail: (name: string) => `${BASE_URL}/pokemon/${name}`,
} as const;

const IMAGES_URL = (name: string) =>
  `https://img.pokemondb.net/sprites/black-white/anim/normal/${name}.gif`;

const LOGO_URL =
  'https://upload.wikimedia.org/wikipedia/commons/9/98/International_Pok%C3%A9mon_logo.svg';

const BREAKPOINTS = {
  small: '576px',
  medium: '768px',
  large: '992px',
  xlarge: '1200px',
};

export { API_ENDPOINTS, BASE_URL, IMAGES_URL, LOGO_URL, BREAKPOINTS };
