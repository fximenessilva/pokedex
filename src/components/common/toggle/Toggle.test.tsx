import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import ToggleButton from './Toggle';
import { COLORS } from '../../../utlils/constants';
import '@testing-library/jest-dom/extend-expect';

describe('ToggleButton', () => {
  it('renders correctly with the default active state', () => {
    const { container } = render(
      <ToggleButton
        onClick={() => {
          //placeholder comment
        }}
        defaultValue={true}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with the default inactive state', () => {
    const { container } = render(
      <ToggleButton
        onClick={() => {
          //placeholder comment
        }}
        defaultValue={false}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('calls the onClick callback when clicked', () => {
    const onClickMock = jest.fn();

    const { getByTestId } = render(
      <ToggleButton onClick={onClickMock} defaultValue={true} />
    );

    const toggleButton = getByTestId('toggle-button');

    fireEvent.click(toggleButton);

    expect(onClickMock).toHaveBeenCalledWith(false);
  });
});
