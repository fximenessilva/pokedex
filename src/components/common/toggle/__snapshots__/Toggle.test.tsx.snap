// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton renders correctly with the default active state 1`] = `
<div
  class="sc-beyTiQ lhUiIW"
  data-testid="toggle-button"
  id="toggle-button"
>
  <div
    class="sc-guDMob eNKguM"
    data-testid="slider"
  />
</div>
`;

exports[`ToggleButton renders correctly with the default inactive state 1`] = `
<div
  class="sc-beyTiQ lhUiIW"
  data-testid="toggle-button"
  id="toggle-button"
>
  <div
    class="sc-guDMob bKbxMq"
    data-testid="slider"
  />
</div>
`;
